packages/mdc-ripple/README.md

```diff
@@ -78,61 +46,81 @@ CSS Class | Description
 
 #### Ripple Mixins
 
 Mixin | Description
 --- | ---
-`mdc-ripple-surface` | Adds base styles for a ripple surface
-`mdc-ripple-radius($radius)` | Adds styles for the radius of the ripple effect,<br>for both bounded and unbounded ripples
+`mdc-ripple-surface` | Mandatory. Adds base styles for a ripple surface
+`mdc-ripple-radius($radius)` | Mandatory. Adds styles for the radius of the ripple effect,<br>for both bounded and unbounded ripples
 
 #### Basic States Mixins
 
-When using the basic states mixins, inclusion of the `mdc-states` mixin is mandatory.
-Inclusion of `mdc-states-activated` or `mdc-states-selected` is optional, depending on whether activated or selected
-states are applicable to the component in question.
-
 Mixin | Description
 --- | ---
-`mdc-states($color, $has-nested-focusable-element)` | Adds state and ripple styles using the indicated color, deciding opacities based on whether the passed color is light or dark.<br>`$has-nested-focusable-element` defaults to `false` but should be set to `true` if the component contains a focusable element (e.g. an input) under the root node.
-`mdc-states-activated($color, $has-nested-focusable-element)` | Adds state and ripple styles for activated states using the indicated color, deciding opacities based on whether the passed color is light or dark.<br>`$has-nested-focusable-element` defaults to `false` but should be set to `true` if the component contains a focusable element (e.g. an input) under the root node.
-`mdc-states-selected($color, $has-nested-focusable-element)` | Adds state and ripple styles for selected states using the indicated color, deciding opacities based on whether the passed color is light or dark.<br>`$has-nested-focusable-element` defaults to `false` but should be set to `true` if the component contains a focusable element (e.g. an input) under the root node.
+`mdc-states($color, $has-nested-focusable-element)` | Mandatory. Adds state and ripple styles in the given color
+`mdc-states-activated($color, $has-nested-focusable-element)` | Optional. Adds state and ripple styles for activated states in the given color
+`mdc-states-selected($color, $has-nested-focusable-element)` | Optional. Adds state and ripple styles for selected states in the given color
+
+> _NOTE_: Each of the mixins above adds ripple styles using the indicated color, deciding opacity values based on whether the passed color is light or dark. `$has-nested-focusable-element` defaults to `false` but should be set to `true` if the component contains a focusable element (e.g. an input) inside the root element.
 
 #### Advanced States Mixins
 
-When using the advanced states mixins, every one of the mixins below should be included at least once to establish base
-styles for states.
+When using the advanced states mixins instead of the basic states mixins, every one of the mixins below should be included at least once.
```

Comment:
Is that true? what happens if I don't specify `mdc-states-hover-opacity($opacity)`? Is it "at least once" or "exactly once"

Reply:
It's "at least once" in the sense that you need to include each mixin once to establish an opacity for that state at all, otherwise there'd be no overlay styles for that state at all. You can potentially use it multiple times for variants (e.g. raised buttons with light ink get different opacities than flat buttons with dark ink).